src/api-binder/index.ts

```diff
@@ -265,13 +265,27 @@ export async function provisionDependentDevice(
 	).timeout(conf.apiTimeout)) as Device;
 }
 
-export function startCurrentStateReport() {
+export async function startCurrentStateReport() {
 	if (balenaApi == null) {
 		throw new InternalInconsistencyError(
 			'Trying to start state reporting without initializing API client',
 		);
 	}
-	startReporting();
+	// Get interval for sending current state
+	const reportInterval = await config.get('stateReportInterval');
+
+	// Store reporting cancelation if reportInterval changes
+	let reporting = await startReporting(reportInterval);
+
+	// If stateReportInterval changed then reset reporting interval
+	config.on('change', async (changedConfig: any) => {
+		if (changedConfig.stateReportInterval != null) {
+			// Clear previous reporting state
+			reporting.cancel();
+			// Set reporting again with new value
+			reporting = await startReporting(changedConfig.stateReportInterval);
```

Comment:
From my manual testing I saw that if we cancel the reporting and start again than a report is sent right away. We may want to change this so that when cancelled we can somehow ensure the next report started respected the old or new interval (probably new interval)

Reply:
The product call decided to keep this behavior of sending a report right away https://jel.ly.fish/brainstorm-topic-6597323c-d100-4205-a257-7713735c0a58